### PR TITLE
Handle duplicate Bundle entries

### PR DESCRIPTION
**Pull request by kaiku**
## Why
Some C-CDA documents repeat the same source act in multiple sections. Stable ID generation maps those repeated identifiers to the same FHIR id, which can produce duplicate Bundle `fullUrl` keys.

## What
- Add duplicate Bundle entry handling during final Bundle assembly.
- Default to `skip`, where the first entry wins and later duplicates are skipped with a warning.
- Add strict `fail` behavior through `duplicate_bundle_entry_policy="fail"`, raising `DuplicateBundleEntryError`.
- Add regression tests for both skip and fail behavior using a synthetic duplicate referral fixture.

## Testing
- `uv run pytest tests/integration/test_athena_ccd.py tests/integration/test_referral_simple.py -q`
- `uv run ruff check ccda_to_fhir/convert.py tests/integration/test_referral_simple.py`
- `uv run ruff format --check ccda_to_fhir/convert.py tests/integration/test_referral_simple.py`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "ccda-to-fhir"
-version = "0.2.22"
+version = "0.2.23"
```
Comment (nt1) [inline pyproject.toml]: I think this change deserves a minor version bump putting us at 0.3.0

**Review by nt1 — APPROVED** [review]
